packages/splunk/data_stream/alert/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,1885 @@
+  - append:
+      field: tags
+      value: preserve_original_event
+      allow_duplicates: false
+      if: ctx.error?.message != null
+  - remove:
+      field:
+        - splunk.alert.annotations.mitre_attack.mitre_tactic
+        - splunk.alert.annotations.mitre_attack.mitre_tactic_id
+        - splunk.alert.annotations.mitre_attack.mitre_technique
+        - splunk.alert.annotations.mitre_attack.mitre_technique_id
+        - splunk.alert.city
+        - splunk.alert.creation_timestamp
+        - splunk.alert.dest
+        - splunk.alert.dest_ip
+        - splunk.alert.dest_port
+        - splunk.alert.device
+        - splunk.alert.devid
+        - splunk.alert.domain
+        - splunk.alert.dst_location
+        - splunk.alert.dstip
+        - splunk.alert.dstport
+        - splunk.alert.email
+        - splunk.alert.file_hash
+        - splunk.alert.file_name
+        - splunk.alert.file_path
+        - splunk.alert.file_size
+        - splunk.alert.file_type
+        - splunk.alert.host
+        - splunk.alert.hostname
+        - splunk.alert.id
+        - splunk.alert.ip
+        - splunk.alert.md5
+        - splunk.alert.os
+        - splunk.alert.os_family
+        - splunk.alert.os_version
+        - splunk.alert.protocol
+        - splunk.alert._raw
+        - splunk.alert.reason
+        - splunk.alert.sha256
+        - splunk.alert.src
+        - splunk.alert.src_ip
+        - splunk.alert.src_location
+        - splunk.alert.src_port
+        - splunk.alert.src_user_email
+        - splunk.alert.src_user_identity_id
+        - splunk.alert.src_user_nick
+        - splunk.alert.srcip
+        - splunk.alert._time
+        - splunk.alert.url
+        - splunk.alert.user
+        - splunk.alert.user_email
+        - splunk.alert.user_identity_id
+        - splunk.alert.userip
+        - splunk.alert.username
+      tag: remove_custom_duplicate_fields
+      ignore_missing: true
+      if: ctx.tags == null || !(ctx.tags.contains('preserve_duplicate_custom_fields'))
+  - remove:
+      field:
+        - _temp
+        - tmp_host
+        - temp_ip
+        - _temp_dest
+      tag: remove_temp
+      ignore_missing: true
+  - script:
+      tag: script_to_drop_null_values
+      lang: painless
+      source: |-
+        boolean drop(Object object) {
+          if (object == null || object == '') {
+            return true;
+          } else if (object instanceof Map) {
+            ((Map) object).values().removeIf(v -> drop(v));
+            return (((Map) object).size() == 0);
+          } else if (object instanceof List) {
+            ((List) object).removeIf(v -> drop(v));
+            return (((List) object).length == 0);
+          }
+          return false;
+        }
+        drop(ctx);
+      description: Drops null/empty values recursively.
+  - set:
+      field: event.kind
+      tag: set_pipeline_error_to_event_kind
+      value: pipeline_error
+      if: ctx.error?.message != null
```

```suggestion
  - set:
      field: event.kind
      tag: set_pipeline_error_to_event_kind
      value: pipeline_error
      if: ctx.error?.message != null
  - append:
      field: tags
      value: preserve_original_event
      allow_duplicates: false
      if: ctx.error?.message != null
```
Essentially, we want to mimic the behaviour of the `on_failure` handler for cases where we have added an error message explicitly in the pipeline. This includes setting the preserve_original_event tag.